pkg/reconciler/openshift/tektonaddon/tektonaddon.go

```diff
@@ -196,11 +205,55 @@ func addPipelineTemplates(manifest *mf.Manifest) error {
 	return tektonaddon.GeneratePipelineTemplates(addonLocation, manifest)
 }
 
-func applyAddons(manifest *mf.Manifest) error {
+func applyAddons(manifest *mf.Manifest, comp v1alpha1.TektonComponent) error {
 	koDataDir := os.Getenv(common.KoEnvKey)
-	addonLocation := filepath.Join(koDataDir, "tekton-addon")
+	addonLocation := filepath.Join(koDataDir, "tekton-addon/"+common.TargetVersion(comp)+"/addons")
+	return appendManifest(manifest, addonLocation)
+}
+
+func applyOptionalAddons(ctx context.Context, manifest *mf.Manifest, comp v1alpha1.TektonComponent, cfg *rest.Config) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+	consoleYamlCRDinstalled, err := isCRDExist(ctx, cfg, consoleYamlSamples)
+	if err != nil {
+		return err
+	}
+	if consoleYamlCRDinstalled {
+		optionalLocation := filepath.Join(koDataDir, "tekton-addon/"+common.TargetVersion(comp)+"/optional/samples")
+		if err := appendManifest(manifest, optionalLocation); err != nil {
+			return err
+		}
+	}
+	consoleQuickStartCRDinstalled, err := isCRDExist(ctx, cfg, consoleQuickStart)
+	if err != nil {
+		return err
+	}
+	if consoleQuickStartCRDinstalled {
+		optionalLocation := filepath.Join(koDataDir, "tekton-addon/"+common.TargetVersion(comp)+"/optional/quickstarts")
+		return appendManifest(manifest, optionalLocation)
+	}
+	return nil
+}
+
+func isCRDExist(ctx context.Context, config *rest.Config, crdName string) (bool, error) {
+	apiextensionsclientset, err := apiextensionsclient.NewForConfig(config)
+	if err != nil {
+		return false, err
+	}
+	_, err = apiextensionsclientset.ApiextensionsV1beta1().
+		CustomResourceDefinitions().Get(ctx, crdName, metav1.GetOptions{})
+	return err == nil, ignoreNotFound(err)
+}
+
+func ignoreNotFound(err error) error {
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
```

Comment:
could we avoid the `ignoreNotFound` function ..
how about,
```
...
_, err = apiextensionsclientset.ApiextensionsV1beta1().
		CustomResourceDefinitions().Get(ctx, crdName, metav1.GetOptions{})
        if err == nil {
           	return true, nil
       }
        if !errors.IsNotFound(err) {
           return false, nil
       }
	return false, err
}
```

Reply:
I feel instead of adding multiple if conditions calling separate functions will be more clear
WDYT?